service/frontend/workflow_handler.go

```diff
@@ -3007,77 +3015,64 @@ func (wh *WorkflowHandler) DescribeSchedule(ctx context.Context, request *workfl
 		}
 	}
 
-	// then query to get current state from the workflow itself
-	// TODO: turn the refresh path into a synchronous update so we don't have to retry in a loop
-	sentRefresh := make(map[commonpb.WorkflowExecution]struct{})
-	// limit how many signals we send, separate from the retry policy (which is used to retry
-	// the query if the signal was not received or processed yet)
-	signalsLeft := 1
 	var queryResponse schedspb.DescribeResponse
 
-	op := func(ctx context.Context) error {
-		req := &historyservice.QueryWorkflowRequest{
-			NamespaceId: namespaceID.String(),
-			Request: &workflowservice.QueryWorkflowRequest{
-				Namespace: request.Namespace,
-				Execution: execution,
-				Query:     &querypb.WorkflowQuery{QueryType: scheduler.QueryNameDescribe},
-			},
-		}
-		res, err := wh.historyClient.QueryWorkflow(ctx, req)
-		if err != nil {
-			return err
-		}
+	req := &historyservice.QueryWorkflowRequest{
+		NamespaceId: namespaceID.String(),
+		Request: &workflowservice.QueryWorkflowRequest{
+			Namespace: request.Namespace,
+			Execution: execution,
+			Query:     &querypb.WorkflowQuery{QueryType: scheduler.QueryNameDescribe},
+		},
+	}
+	res, err := wh.historyClient.QueryWorkflow(ctx, req)
+	if err != nil {
+		return nil, err
+	}
 
-		queryResponse.Reset()
-		err = payloads.Decode(res.GetResponse().GetQueryResult(), &queryResponse)
-		if err != nil {
-			return err
-		}
+	queryResponse.Reset()
+	err = payloads.Decode(res.GetResponse().GetQueryResult(), &queryResponse)
+	if err != nil {
+		return nil, err
+	}
 
-		// Search attributes in the Action are already in external ("aliased") form. Do not alias them here.
+	// Search attributes in the Action are already in external ("aliased") form. Do not alias them here.
 
-		// for all running workflows started by the schedule, we should check that they're
-		// still running, and if not, poke the schedule to refresh
-		needRefresh := false
-		for _, ex := range queryResponse.GetInfo().GetRunningWorkflows() {
-			if _, ok := sentRefresh[*ex]; ok {
-				// we asked the schedule to refresh this one because it wasn't running, but
-				// it's still reporting it as running
-				return errWaitForRefresh
-			}
+	runningWorkflows := make([]*commonpb.WorkflowExecution, 0, len(queryResponse.GetInfo().GetRunningWorkflows()))
```

Comment:
```suggestion
	var runningWorkflows []*commonpb.WorkflowExecution
```
The common case here is 0, sometimes 1, so no need for make